packages/db/postgres/migrations/020-delete-remote-pins.sql

```diff
@@ -0,0 +1,17 @@
+-- Ran a few stats in staging (data should be comparable to prod)
+-- # pins 198.607.016
+-- # Remote pins: 131.495.153
+-- % To be removed: ~66%
+
+-- Concern: Is deleting a few hundred millions rows in Prod affecting performance?
+-- Research:
+-- According to documentation https://www.postgresql.org/docs/7.2/locking-tables.html deleting rows should use a 
+-- `RowExclusiveLock`. 
+-- This could have a negative performance impact in case those rows are being updated by other queries.
+-- However in our specific case I don't think remote pin are being really ever touched so it should be safe to just run 
+-- the query.
+-- Not sure about what statement_timeout we have in prod? Could that be a problem?
+--
+-- Alternative approach would be to create a cron job and delete in batches.
+
+DELETE FROM pin WHERE status='Remote'
```

Comment:
Still need to look into/think about cascading deletes

Reply:
In terms of performance, the `status` column has got an index, and like you say, these rows are unlikely to be being touched by other processes. But my concern is about how much strain that single `DELETE` statement puts on the DB instance/memory/CPU in one go. from the docs:

> PostgreSQL actually treats every SQL statement as being executed within a transaction. If you do not issue a BEGIN command, then each individual statement has an implicit BEGIN and (if successful) COMMIT wrapped around it.

I don't really know how postgres transactions work under the hood, but I imagine that a single operation on millions of rows _might_ be holding quite a lot of stuff in memory, and so it might just hog the resources of the machine, even if there isn't any conflict on the rows that it's touching.

So I'm kind of leaning towards the cron job idea.